lib/pbench/server/api/resources/__init__.py

```diff
@@ -649,30 +661,38 @@ def _check_authorization(self, user: str, access: str):
                 access.
         """
         authorized_user: User = Auth.token_auth.current_user()
-        authorized = True
         self.logger.debug(
             "Authorizing {} access for {} to user {} with access {}",
             self.role,
             authorized_user,
             user,
             access,
         )
+
+        # Default user to the authenticated user, if not specified
+        if authorized_user and not user:
+            user = authorized_user.username
```

Comment:
This doesn't affect what the query builder will see, only authorization. If `"user"` is missing in the query builder, and you have admin or you're looking for public data, we'll just build the `authorization.access` term. But if you're looking for private data (without a username) and don't have admin, it'll build "user == me and access == private"; so the check here is appropriate and avoids dealing with a missing `"user"` later.

Probably needs a better explanation, though. ☹️ Basically, if you don't specify user, either you're admin and we validate your undisputed right to the entire multiverse or we validate your right to only your own data because we decided to constrain the query rather than failing.